### PR TITLE
Feature/add testing

### PR DESCRIPTION
**Pull request by denningk**
# Feature/Fix Name: 

## Description
Add testing to the project. Run type-check, linting, and testing in pipeline automatically. Tests can be run locally with `npm run test`.
## Changes
  - Configured Jest for testing
  - Added ts-jest for typescript testing
  - Adding testing to `.gitlab-ci.yml` along with type-checking and linting
## Type of change
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update
### All Submissions:
* [ ] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [ ] Any dependent changes have been merged and published in downstream modules

### New Feature Submissions:

1. [x] Does your submission pass tests?

2. [x] Have you lint your code locally prior to submission?

### Changes to Core Features:
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

* [x] Have you written new tests for your core changes, as applicable?

* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -5,17 +5,19 @@ stages:
...
-  key: ${CI_COMMIT_REF_SLUG}
-  paths:
-    - node_modules/
 
 build_site:
   image: node:10.16.0
   stage: build
   script:
     - npm install
+    - npm run type-check
```
Comment (kevincianfarini) [inline .gitlab-ci.yml:11]: these two should probably be in their own `build_lint` job. That way, you can only run linting on merge request and not waste it on merges to master and such.

**Review by kevincianfarini — COMMENTED** [review]
I don't know what 99% of this means.

File: .gitlab-ci.yml (hunk 1 above)
Comment (denningk) [reply .gitlab-ci.yml:11]: Good call. I'll change that

**Comment by denningk** [comment]
Most of it is just test configuration. I created mocks for random stuff like Gatsby itself so Jest can use it.

**Review by kevincianfarini — APPROVED** [review]
lGTM

**Pull request merged**